include/crypto/ciphermode_platform.h

```diff
@@ -12,7 +12,8 @@
 
 # include "openssl/aes.h"
 
-# ifdef VPAES_ASM
+# ifndef OPENSSL_NO_DEPRECATED
```

Comment:
Not quite sure I understand why the NO_DEPRECTAED is required in this internal file?
(This file is used also inside provider implementations).

Reply:
This file is included by e_camellia.c which does not include the "internal/deprecated.h" file, because it doesn't use any externally deprecated APIs. However this header file contains references to AES things that may not be defined in a no-deprecated build - and so the compiler complains loudly. Adding this guard means e_camellia.c compiles correctly. Another fix would be to instead include "internal/deprecated.h" in e_camellia.c - but I'd rather not do that.

Reply:
cipher_platform.h should really be divided into algo specific headers.

Reply:
Maybe. But I'd see that as outside of the scope of this PR.

Reply:
#10662

Reply:
Now that #10662 has gone in, this file no longer exists and these changes aren't necessary.